### PR TITLE
Added Dodge the Blocks game

### PR DESCRIPTION
**Pull request by RudraJy**
### Requesting to submit a pull request to the PyVerse repository.

---

#### Issue Title
**Please enter the title of the issue related to your pull request.**  
[Code Addition Request]: Add Game Development Project : Dodge the Blocks

- [x] I have provided the issue title.

---

#### Info about the Related Issue
**What's the goal of the project?**  
To add a Dodge the Blocks game created in Python using Pygame.

- [x] I have described the aim of the project.

---

#### Name
**Please mention your name.**  
Rudra Jyotirmay

- [x] I have provided my name.

---

#### GitHub ID
**Please mention your GitHub ID.**  
RudraJy

- [x] I have provided my GitHub ID.

---

#### Email ID
**Please mention your email ID for further communication.**  
jyot.rud2003@gmail.com

- [x] I have provided my email ID.

---

#### Identify Yourself
**Mention in which program you are contributing (e.g., WoB, GSSOC, SSOC, SWOC).**  
Contributor - Hacktoberfest 2024
Contributor - GSSOC 24 : Extended Edition

- [x] I have mentioned my participant role.

---

#### Closes
**Enter the issue number that will be closed through this PR.**  
*Closes: #714 

- [x] I have provided the issue number.

---

#### Describe the Add-ons or Changes You've Made
**Give a clear description of what you have added or modified.**  
Added a game in the Game_Development folder.

- [x] I have described my changes.

---

#### Type of Change
**Select the type of change:**  
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

---

#### How Has This Been Tested?
**Describe how your changes have been tested.**  
Tested on my local system.

- [x] I have described my testing process.

---

#### Checklist
**Please confirm the following:**  
- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: Game_Development/Dodge-the-Blocks/main.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+
+    # Collision detection
+    blocks_hit_list = pygame.sprite.spritecollide(player, block_list, True)
+    all_sprites_list.draw(screen)
+    pygame.display.flip()
+
+
+    # Player movement
+    handle_player_movement(event, player)
```
Comment (ruhi47) [inline Game_Development/Dodge-the-Blocks/main.py:206]: The function ```handle_player_movement(event, player)``` is called outside the event loop. Due to this only last event is repeating (i.e. Start of the game) and it crashes afterwards. Please update this and will merge once updated.

File: Game_Development/Dodge-the-Blocks/main.py [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+pygame.display.set_caption("My Game")
+
+pygame.mixer.music.load(welcome_screen_music)
+start_screen(screen)
+pygame.mixer.music.stop()
+pygame.mixer.music.load(in_game_music)
+pygame.mixer.music.play(-1)
+
+# Main game loop and logic
+pygame.init()
```
Comment (ruhi47) [inline Game_Development/Dodge-the-Blocks/main.py:144]: Don't need this twice. Initialisation required just at the start.

**Review by ruhi47 — CHANGES_REQUESTED** [review]

**Review by TheChaoticor — CHANGES_REQUESTED** [review]
Please do the requested changes

File: Game_Development/Dodge-the-Blocks/main.py (hunk 2 above)
Comment (RudraJy) [reply Game_Development/Dodge-the-Blocks/main.py]: Removed second pygame.init(). Thanks for the correction

File: Game_Development/Dodge-the-Blocks/main.py (hunk 1 above)
Comment (RudraJy) [reply Game_Development/Dodge-the-Blocks/main.py:205]: That won't work. When the `handle_player_movement(event, player)` function is called inside the event loop, it only processes discrete events. Pygame doesn’t continuously repeat key events while a key is held down, but rather treats it as individual key presses that need to be handled explicitly.

So, what would happen if the function is called inside the event loop?
Continuing to press on an arrow key only moves it by one unit and then it stops. For example if i hold down the _down arrow key_ then the player block moves one unit down and then stops. If i want to move further down, i have to lift up my finger and then press the down key again. This is annoying and not the desired functionality.

Why keep the function outside the event loop?
It keeps checking for the event in every frame, meaning it continually processes movement as long as the key is pressed, which is what you want for smooth movement. This is the desired functionality.

**Pull request closed without merging**